File: codes/sates_test_cs/report/reporter_factory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace sates.test.cs.report
{
    public enum REPORTER_TYPE
    {
        LOCAL_JSON
    }
    public class reporter_factory
    {
        public static reporter create(REPORTER_TYPE reporter_type)
        {
            reporter retval = null;
            if (REPORTER_TYPE.LOCAL_JSON == reporter_type)
            {
                retval = new reporter_local_json();
            }

            return retval;
        }
    }
}

File: codes/sates_test_cpp/sates/report/reporter_factory.cpp
#include <sates/report/reporter_factory.h>
#include <sates/report/reporter_local_json.h>

namespace sates
{
    namespace report
    {
        reporter* reporter_factory::create(REPORTER_TYPE reporter_type)
        {
            reporter* p_retval = nullptr;
            if (REPORTER_TYPE::LOCAL_JSON == reporter_type)
            {
                p_retval = new reporter_local_json();
            }
            return p_retval;
        }
    }
}
